### PR TITLE
[1.8.9] Base code for fixing all machine types

### PR DESCRIPTION
**Pull request by AlexIIL**
[ api | [main](https://github.com/CovertJaguar/Railcraft/pull/802) ]

A small set of changes required for the bigger one.

### CODE REVIEW

File: mods/railcraft/api/core/items/StackFilter.java [hunk 1]
```diff
@@ -113,4 +115,15 @@ public boolean apply(ItemStack stack) {
         };
     }
 
+    public static StackFilter anyOf(@Nonnull final ItemStack[] stacks) {
+        return new StackFilter() {
```
Comment (CovertJaguar) [inline mods/railcraft/api/core/items/StackFilter.java]: This is basically a more awkward ArrayStackFilter yes?

Comment (AlexIIL) [inline mods/railcraft/api/core/items/StackFilter.java]: Yes. And its use case is for an inventory so even more awkward than it should be.

Comment (AlexIIL) [inline mods/railcraft/api/core/items/StackFilter.java]: ( I was going to remove it but you have already pulled it)

Comment (CovertJaguar) [inline mods/railcraft/api/core/items/StackFilter.java]: No worries, I got it.

**Pull request merged**